terraform/bod_docker_iam.tf

```
@@ -0,0 +1,98 @@
+# Create the IAM instance profile for the BOD Docker EC2 server instance
+
+# The instance profile to be used
+resource "aws_iam_instance_profile" "bod_docker" {
+  name = format("bod_docker_instance_profile_%s", local.production_workspace ? "production" : terraform.workspace)
+  role = aws_iam_role.bod_docker_instance_role.name
+}
+
+# The instance role
+resource "aws_iam_role" "bod_docker_instance_role" {
+  name               = format("bod_docker_instance_role_%s", local.production_workspace ? "production" : terraform.workspace)
+  assume_role_policy = data.aws_iam_policy_document.ec2_service_assume_role_doc.json
+}
+
+# Attach the CloudWatch Agent policy to this role as well
+resource "aws_iam_role_policy_attachment" "cloudwatch_agent_policy_attachment_bod_docker" {
+  role       = aws_iam_role.bod_docker_instance_role.id
+  policy_arn = "arn:aws:iam::aws:policy/CloudWatchAgentServerPolicy"
+}
+
+# Attach the SSM Agent policy to this role as well
+resource "aws_iam_role_policy_attachment" "ssm_agent_policy_attachment_bod_docker" {
+  role       = aws_iam_role.bod_docker_instance_role.id
+  policy_arn = "arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore"
+}
+
+# IAM policy document that that allows the invocation of our Lambda
+# functions.  This will be applied to the role we are creating.
+data "aws_iam_policy_document" "lambda_bod_docker_doc" {
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "lambda:InvokeFunction",
+    ]
+
+    # I should be able to use splat syntax here
+    resources = [
+      aws_lambda_function.lambdas[0].arn,
+      aws_lambda_function.lambdas[1].arn,
+      aws_lambda_function.lambdas[2].arn,
+    ]
```

We can get rid of these hard-coded lambda counts:
```suggestion
    resources = [ for lambda in aws_lambda_function.lambdas : lambda.arn ]
```

---

@dav3r I have that taken care of as part of some other work ([improvement/refactor_bod_lambda_configuration](https://github.com/cisagov/cyhy_amis/tree/improvement/refactor_bod_lambda_configuration)). Do you mind if I leave that out of this PR?

---

👍 No problem at all.